### PR TITLE
More cppcheck fixes

### PR DESCRIPTION
**Pull request by crbrox**
Related with https://github.com/telefonicaid/fiware-orion/issues/1368

### CODE REVIEW

File: src/lib/logMsg/logMsg.h [hunk 1]
```diff
@@ -1775,11 +1772,10 @@ do                                                              \
 *
 * LM_TRANSACTION_END -
 */
-#define LM_TRANSACTION_END()                              \
-do                                                        \
-{                                                         \
+inline void LM_TRANSACTION_END()                                                                                     \
+{
   LM_I(("Transaction ended"));                            \
   LM_TRANSACTION_RESET();                                 \
```
Comment (kzangeli) [inline src/lib/logMsg/logMsg.h]: Here you forgot the backslashes at the end of the line

File: src/lib/rest/RestService.cpp [hunk 2]
```diff
@@ -442,8 +442,7 @@ std::string restService(ConnectionInfo* ciP, RestService* serviceV)
     // A tenant string must not be longer than 50 characters and may only contain
     // underscores and alphanumeric characters.
     //
-    std::string result;
-    if ((ciP->tenant != "") && ((result = tenantCheck(ciP->tenant)) != "OK"))
+    if ((ciP->tenant != "") && (tenantCheck(ciP->tenant)) != "OK")
     {
       OrionError  error(SccBadRequest,
                         "tenant name not accepted - a tenant string must not be longer than " MAX_TENANT_NAME_LEN_STRING " characters"
```
Comment (kzangeli) [inline src/lib/rest/RestService.cpp]: This here seems like a mistake.
"result" should have been used, and not a fixed string (as detail for OrionError).
I just looked at tenantCheck() and (as I suspected) it has two ways of failing (invalid char + name too long).
Let's use this knowledge and fix this error, returning the correct error string (result).

File: src/lib/logMsg/logMsg.h (hunk 1 above)
Comment (kzangeli) [inline src/lib/logMsg/logMsg.h]: Ah, and also, please put in 'void' inside '()'   (undocumented style guide)

Comment (crbrox) [inline src/lib/logMsg/logMsg.h]: Fixed eb662e3

File: src/lib/common/string.cpp [hunk 3]
```diff
@@ -95,7 +95,6 @@ bool isIPv6(const std::string& in)
 {
   size_t      pos;
   std::string partip;
-  std::string resu;
```
Comment (fgalan) [inline src/lib/common/string.cpp]: I understand that this is a case of variable not used, detected by cppcheck.

The code is supposed to be compiled with gcc flags to fail with error when variable not used is detected. I wonder why the compiler has not complaint about this...

Anyway, NTC, good catches.

Comment (kzangeli) [inline src/lib/common/string.cpp]: Look a little further down.  It is added to and after that not used.
cppcheck seems a bit smarter than the compiler :-)

**Comment by fgalan** [comment]
LGTM

File: src/lib/common/string.cpp (hunk 3 above)
Comment (fgalan) [inline src/lib/common/string.cpp]: > cppcheck seems a bit smarter than the compiler :-)

I'm scared... xD

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**